### PR TITLE
Fix the schema change errors and some possible vulnerabilities

### PR DESCRIPTION
**Pull request by pastakhov**

### CODE REVIEW

File: includes/RottenLinksPager.php [hunk 1]
```diff
@@ -37,14 +38,18 @@ public function formatValue( $name, $value ) {
...
 			case 'rl_pageusage':
 				$number = count( json_decode( $row->rl_pageusage, true ) );
-				$formatted = "<a href=\"{$this->config->get( 'ScriptPath' )}/index.php?title=Special%3ALinkSearch&target={$row->rl_externallink}\">{$number}</a>";
+				$specialLinkSearch = SpecialPage::getTitleFor( 'LinkSearch' );
+				$href = $specialLinkSearch->getInternalURL( [ 'target' => $row->rl_externallink ] );
+				$formatted = HTML::element( 'a', [ 'href' => $href ], $number );
 				break;
 			default:
-				$formatted = "Unable to format $name";
+				$formatted = HTML::element( 'sapn', [], "Unable to format $name" );
```
Comment (paladox) [inline includes/RottenLinksPager.php:52]: Typo :)

File: includes/RottenLinksPager.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 class RottenLinksPager extends TablePager {
 	private $config = null;
+	private $showBad;
```
Comment (paladox) [inline includes/RottenLinksPager.php:7]: This seems unused.

Comment (paladox) [reply includes/RottenLinksPager.php:7]: Oh nvm, sorry :)

File: includes/RottenLinksPager.php (hunk 1 above)
Comment (pastakhov) [reply includes/RottenLinksPager.php]: fixed :)

Comment (paladox) [reply includes/RottenLinksPager.php]: Thanks!

**Comment by paladox** [comment]
Hi, would you be able to send us what the security issue is to security((a))miraheze.org please?

**Comment by pastakhov** [comment]
> Hi, would you be able to send us what the security issue is to security((a))miraheze.org please?

never write code like replaced in includes/RottenLinksPager.php, it always opens a door for HTML injection. I was not looking for any specific examples of how this can be exploited. In any case, this is bad practice and I think that if you want, you can find an exploit

**Pull request merged**